### PR TITLE
feat: Toolbar 3000 changes

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Following up on Paul's great work with some tweaks and changes that I believe polish it off

## Changes

* Updates the styles to be (I believe) a little nicer and more 3000
* Removes the drag and drop for our own impl which gives us way more control
* Allows the popover to be bigger than the bar

## TODO
- [ ] Fix clicking button whilst dragging still triggers button
- [x] Fix it so that when you un-minimize, it opens the previous panel / active state

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/toolbar/stats/HeatmapToolbarMenu.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ const MenuBody = (): JSX.Element => {
                     />
                 </div>
             </Tooltip>
-            <div className="FlagList flex flex-col w-full overflow-y-scroll h-full">
+            <div className="FlagList flex flex-col w-full h-full">
```
Comment (pauldambra) [inline frontend/src/toolbar/stats/HeatmapToolbarMenu.tsx:78]: I preferred these menus with a separate not-scrolling header but would rather get it all merged than not 🤷

File: frontend/src/toolbar/toolbarLogic.ts [hunk 2]
```diff
@@ -65,8 +65,8 @@ export const toolbarLogic = kea<toolbarLogicType>([
         },
         processUserIntent: async () => {
             if (props.userIntent === 'add-action' || props.userIntent === 'edit-action') {
```
Comment (pauldambra) [inline frontend/src/toolbar/toolbarLogic.ts:67]: Yeah, I don't know how you would launch the toolbar and edit an action at the same time. I vote to delete it and then add it back if someone can tell us what broke

**Review by pauldambra — APPROVED** [review]

File: frontend/src/toolbar/button/HedgehogButton.tsx [hunk 3]
```diff
@@ -32,9 +31,10 @@ export function HedgehogButton(): JSX.Element {
                     onClose={() => {}}
                     actorRef={actorRef}
                     isDarkModeOn={theme === 'dark'}
-                    onPositionChange={(actor) => {
-                        saveDragPosition(actor.x + SPRITE_SIZE, -actor.y - SPRITE_SIZE)
+                    onPositionChange={() => {
+                        syncWithHedgehog()
```
Comment (daibhin) [inline frontend/src/toolbar/button/HedgehogButton.tsx:35]: The most PostHog function name ever

**Review by daibhin — COMMENTED** [review]
Feels good. One tiny nit is that the scaling transition when dragging feels like it could be sped up a bit to me, certainly after dropping (I have less of an opinion on drag start). Kind of falls under this thinking for me https://posthog.slack.com/archives/C064U4XCPHQ/p1699987726918269

**Review by daibhin — APPROVED** [review]

File: frontend/src/toolbar/stats/HeatmapToolbarMenu.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/toolbar/stats/HeatmapToolbarMenu.tsx:78]: I do to but it was getting tough to get right on a smaller screen with all the possible flexing that can happen

File: frontend/src/layout/navigation-3000/themeLogic.ts
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/layout/navigation-3000/themeLogic.ts:1]: Sort of but no. I'm not sure why but now and then I could get mounted logic errors where as when I added this they went away. Something aboout the new order of files being imported...

File: frontend/__snapshots__/scenes-other-toolbar-components-3000--actions-dark-mode.png
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/__snapshots__/scenes-other-toolbar-components-3000--actions-dark-mode.png:1]: Fixed these!

File: frontend/src/toolbar/button/HedgehogButton.tsx (hunk 3 above)
Comment (benjackwhite) [reply frontend/src/toolbar/button/HedgehogButton.tsx:35]: :D

**Pull request merged**